lib/Auth/Source/Negotiate.php

```diff
@@ -166,7 +170,16 @@ public function authenticate(array &$state): void
                 }
             }
 
-            $auth = new \KRB5NegotiateAuth($this->keytab);
+            if ($this->spn === null) {
+                // old FQDN behavior
+                $auth = new \KRB5NegotiateAuth($this->keytab);
+            } elseif($this->spn == '0') {
```

Comment:
This could be simplified and just forward $this->spn if it's not null, I had this specifically to avoid explaining ops people the difference between '0' and 0.